### PR TITLE
Custom legend jumps

### PR DESCRIPTION
**Pull request by mmcfarland**
This is a stopgap measure to prevent legend flickering when using the
Habitat Explorer.  The comparison of layers on the map vs legends in the
Legend is expanded to account for possible custom legend.  The need to do
the comparison is a result of the existing unreliability of the JSAPI to
notify of `layerAdd` events.

Connects #502 

#### To test:

* Check that the legend does not flicker in the development build.
* Check that the future habitat plugin in split screen does not flicker (attached)
* Check that after turning on and turning off the Community Rating System plugin, there is no flicker.

#### Test Plugins

(Make sure these extract like `plugins/future_habitat/main.js`, the zip may double up the plugin directory path)
* [future_habitat.zip](https://github.com/CoastalResilienceNetwork/GeositeFramework/files/116013/future_habitat.zip)
* [community-rating-system.zip](https://github.com/CoastalResilienceNetwork/GeositeFramework/files/116309/community-rating-system.zip)

### CODE REVIEW

File: src/GeositeFramework/js/Map.js [hunk 1]
```diff
@@ -307,8 +315,13 @@ require(['use!Geosite',
                 return !layer.graphics && layer.visible;
             });
 
-            // Subtract one for the base layer.
-            if ((totalVisibleLayers.length - 1) !== nuggetCount) {
+            // Subtract one for the base layer and one for each 
+            // custom legend that is visible and has children
```
Comment (mmcfarland) [inline src/GeositeFramework/js/Map.js]: I'll rebase a change to this comment that includes the description of the `serviceOptOut` after review.

**Comment by caseypt** [comment]
+1. Tested. There is some initial flickering when layers are first added, but in all cases I tested, things settle down in a second or two.

**Comment by mmcfarland** [comment]
Yeah, I noticed that too, do you think it's worth increasing the redraw delay time?

**Comment by caseypt** [comment]
Yeah, that might be a simple way to get around it.

**Comment by mmcfarland** [comment]
Increased from `250 -> 750` in https://github.com/mmcfarland/GeositeFramework/commit/009cf862b504861748e4e5f484f0e1c31f40256c

I don't notice flickering any more.

**Comment by caseypt** [comment]
Just did a couple quick tests and flickering has been significantly reduced.

**Comment by mmcfarland** [comment]
Cool, thanks for the review.  Merging.

**Pull request merged**